cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go

```diff
@@ -43,7 +53,49 @@ func (r *ImageMapReader) Read(ctx context.Context) ([]common.ImageMapItem, error
 		return []common.ImageMapItem{}, err
 	}
 
-	return []common.ImageMapItem{}, nil // TODO: Implement.
+	err = r.readBAT(ctx)
+	if err != nil {
+		return []common.ImageMapItem{}, err
+	}
+
+	var items []common.ImageMapItem
+	var item common.ImageMapItem
+	var entry imageMapEntry
+
+	for i, _ := range r.bat {
```

Comment:
The point of my proposal was to use r.bat iterator to access the address directly, without the necessity to use the index. Using the index if you have an iterator just looks ugly, despite the fact that I understand that the index is within the array range.